assets/src/state/pagePanelState.ts

```diff
@@ -0,0 +1,218 @@
+    [PagePath.SearchMaps]: initialPageState,
+  },
+}
+
+export type VehicleType = Vehicle | Ghost | null | undefined
+
+//#region Reducers
+export const openViewReducer = (
+  currentState: ViewState,
+  action: Action
+): ViewState => {
+  let currentPath = currentState.currentPath
+  if (action.type === "SET_CURRENT_PATH") {
+    currentPath = action.path
+  }
+
+  return {
+    currentPath,
+    state: {
+      ...currentState.state,
+      [currentState.currentPath]: openViewPanelReducer(
+        currentState.state[currentState.currentPath],
+        action
+      ),
+    },
+  }
+}
+
+const openViewPanelReducer = (
+  state: PageViewState,
+  action: Action
+): PageViewState => {
+  const { openView, previousView, selectedVehicleOrGhost } = state
+  switch (action.type) {
+    case "SET_CURRENT_PATH": {
+      return {
+        openView: OpenView.None,
+        previousView: OpenView.None,
+        selectedVehicleOrGhost,
+      }
+    }
+    case "OPEN_NOTIFICATION_DRAWER":
+      return openView === OpenView.NotificationDrawer
+        ? state
+        : {
+            openView: OpenView.NotificationDrawer,
+            previousView: openView,
+            selectedVehicleOrGhost: undefined,
+          }
+    case "OPEN_SWINGS_VIEW":
+      return openView === OpenView.Swings
+        ? state
+        : {
+            openView: OpenView.Swings,
+            previousView: openView,
+            selectedVehicleOrGhost: undefined,
+          }
+    case "OPEN_LATE_VIEW":
+      return openView === OpenView.Late
+        ? state
+        : {
+            openView: OpenView.Late,
+            previousView: openView,
+            selectedVehicleOrGhost: undefined,
+          }
+    case "CLOSE_VIEW":
+      return openView !== null
+        ? {
+            openView: OpenView.None,
+            previousView: OpenView.None,
+            selectedVehicleOrGhost: undefined,
+          }
+        : state
+    case "SELECT_VEHICLE":
+    case "SELECT_VEHICLE_FROM_NOTIFICATION":
+      return {
+        openView: OpenView.None,
+        previousView: openView === OpenView.None ? previousView : openView,
+        selectedVehicleOrGhost: action.payload.vehicle,
+      }
+    case "SET_NOTIFICATION":
+      return {
+        openView,
+        previousView: previousView,
+        selectedVehicleOrGhost: undefined,
+      }
+    case "RETURN_TO_PREVIOUS_VIEW":
+      return previousView !== OpenView.None
+        ? {
+            openView: previousView,
+            previousView: OpenView.None,
+            selectedVehicleOrGhost: undefined,
+          }
+        : state
+    default:
+      return state
+  }
+}
+//#endregion Reducers
+
+//#region Accessors
+// Keeping this around because this is used by other reducers in `state.ts`.
+export const selectVehicleFromNotification = (
+  vehicle: VehicleType
+): SelectVehicleFromNotificationAction => ({
+  type: "SELECT_VEHICLE_FROM_NOTIFICATION",
+  payload: { vehicle },
+})
+//#endregion Accessors
```

Comment:
**question:** What's your rationale for calling this section "Accessors?" I'm not sure exactly what I'd call it, but something more like "action constructors" would make more sense to me personally.

Reply:
It was "simply" the first thing that came to mind when I was migrating 😅 
I like "action constructors"! Though I might shorten to just actions if that works.

Reply:
ffcab85bea8023ac71cc41cb41bb4b32e53e03ed